mistlib/core.py

```diff
@@ -305,58 +305,24 @@ def write_pdf(self, file):
     
     def write_adamantine_input(self, file):
         reference_temperature = self.properties["solidus_eutectic_temperature"].value # For adamantine we assume that all temperature-dependent material properties are evaluated at the solidus temperature
-
-        p = self.properties["specific_heat_solid"]
-        if (p.value_type == ValueTypes.SCALAR):
-                specific_heat_1 = p.value
-        elif (p.value_type == ValueTypes.LAURENT_POLYNOMIAL):
-                specific_heat_1 = p.evaluate_laurent_polynomial(reference_temperature)
-        else:
-                print("Error: adamantine requires either SCALAR or LAURENT_POLYNOMIAL ValueTypes.")
-
-        p = self.properties["specific_heat_liquid"]
-        if (p.value_type == ValueTypes.SCALAR):
-                specific_heat_2 = p.value
-        elif (p.value_type == ValueTypes.LAURENT_POLYNOMIAL):
-                specific_heat_2 = p.evaluate_laurent_polynomial(reference_temperature)
-        else:
-                print("Error: adamantine requires either SCALAR or LAURENT_POLYNOMIAL ValueTypes.")
-
-        p = self.properties["thermal_conductivity_solid"]
-        if (p.value_type == ValueTypes.SCALAR):
-                thermal_conductivity_1 = p.value
-        elif (p.value_type == ValueTypes.LAURENT_POLYNOMIAL):
-                thermal_conductivity_1 = p.evaluate_laurent_polynomial(reference_temperature)
-        else:
-                print("Error: adamantine requires either SCALAR or LAURENT_POLYNOMIAL ValueTypes.")
-
-        density = None
-        p = self.properties["density"]
-        if p.value_type == ValueTypes.SCALAR:
-                density = p.value
-        elif p.value_type == ValueTypes.LAURENT_POLYNOMIAL:
-                density = p.evaluate_laurent_polynomial(reference_temperature)
-        else:
-                print("Error: adamantine requires either SCALAR or LAURENT_POLYNOMIAL ValueTypes.")
-
-        thermal_conductivity = None
-        p = self.properties["thermal_conductivity_liquid"]
-        if p.value_type == ValueTypes.SCALAR:
-                thermal_conductivity_2 = p.value
-        elif p.value_type == ValueTypes.LAURENT_POLYNOMIAL:
-                thermal_conductivity_2 = p.evaluate_laurent_polynomial(reference_temperature)
-        else:
-                print("Error: adamantine requires either SCALAR or LAURENT_POLYNOMIAL ValueTypes.")
-
-        emissivity = None
-        p = self.properties["emissivity"]
-        if p.value_type == ValueTypes.SCALAR:
-                emissivity = p.value
-        elif p.value_type == ValueTypes.LAURENT_POLYNOMIAL:
-                emissivity = p.evaluate_laurent_polynomials(reference_temperature)
-        else:
-                print("Error: adamantine requires either SCALAR or LAURENT_POLYNOMIAL ValueTypes.")
-
+        def get_property(self, property_name,):
```

Comment:
This makes it a local function instead of a class function - if you look at #20 that one function can be reused for all write functions